crossing-repair.lic

```diff
@@ -33,6 +87,77 @@ class CrossingRepair
+    # yaml list of equipment. While having to repair different items at leather vs. metal
+    # repair is deprecated, we still need to support the legacy settings.
+    leather_items_to_repair = @equipment_manager.items.select(&:leather).reject(&:skip_repair)
+    metal_items_to_repair = @equipment_manager.items.reject(&:leather).reject(&:skip_repair)
+
+    # No bank specified in base-town.yaml for specified repair town? Message user. This method
+    # both withdraws and deposits coins, so having a bank specified is requisite.
+    unless @hometown_data['deposit']
+      DRC.message("No bank found for #{town} in base-town.yaml. Please submit an issue on GitHub:")
+      DRC.message("https://github.com/rpherbig/dr-scripts/issues")
+      exit
+    end
+
+    # Check if we have leather items specified in yaml. If so, verify we have a leather
+    # repair person specified in base-town.yaml for the user-specified repair town.
+    # If not, message user and direct to report issue.
+    if leather_items_to_repair.any?
+      echo "Leather items detected." if $debug_mode_crossing_repair
+      unless @hometown_data['leather_repair']
+        DRC.message("No leather repairer specified in base-town.yaml. Please submita an issue on GitHub:")
+        DRC.message("https://github.com/rpherbig/dr-scripts/issues")
+        exit
+      end
+    end
+
+    # Check if we have metal items specified in yaml. If so, verify we have a metal
+    # repair person specified in base-town.yaml for the user-specified repair town.
+    # If not, message user and direct to report issue.
+    if metal_items_to_repair.any?
+      echo "Metal items detected." if $debug_mode_crossing_repair
+      unless @hometown_data['metal_repair']
+        DRC.message("No metal repairer specified in base-town.yaml. Please submita an issue on GitHub:")
+        DRC.message("https://github.com/rpherbig/dr-scripts/issues")
+        exit
+      end
+    end
+
+    # Verify we have enough money as specified in the user's yaml
+    # Also start the performance script for some lovely extra exp = tdps
+    DRCM.ensure_copper_on_hand(@repair_withdrawal_amount, settings)
+    start_script('performance', ['noclean']) unless settings.instrument
+
+    # Display our metal and leather item hashes for repair if debug set
+    if $debug_mode_crossing_repair
+      DRC.message("Metal items hash:")
+      echo @hometown_data['metal_repair'] => metal_items_to_repair
+
+      DRC.message("Leather items hash:")
+      echo @hometown_data['leather_repair'] => leather_items_to_repair
+    end
+
+    # If we've not exited above, we're validated ready to repair.
+    # Populate a hash of items to repair at either leather or metal repairer.
+    # Return this hash for future use.
+    repair_info = { @hometown_data['leather_repair'] => leather_items_to_repair }
+                    .merge(@hometown_data['metal_repair'] => metal_items_to_repair) { |_key, old, new| old + new }
```

Comment:
💡 Stylistically, to reduce some duplicated code and (subjectively) improve readability of the repair info merge statement, what are your thoughts on the following:

```ruby
    metal_repair_info = {
      @hometown_data['metal_repair'] => metal_items_to_repair
    }

    leather_repair_info = {
      @hometown_data['leather_repair'] => leather_items_to_repair
    }

    # Display our metal and leather item hashes for repair if debug set
    if $debug_mode_crossing_repair
      DRC.message("Metal items hash:")
      echo metal_repair_info

      DRC.message("Leather items hash:")
      echo leather_repair_info
    end

    # If we've not exited above, we're validated ready to repair.
    # Populate a hash of items to repair at either leather or metal repairer.
    # Since the same repairer may do both kinds of items, merge the lists together.
    repair_info = [metal_repair_info, leather_repair_info].reduce({}) do |result, info|
      result.merge(info) { |key, old_value, new_value| old_value + new_value }
    end
```

Reply:
@KatoakDR helped on this section.